### DISCUSSION TITLE
How to find package by description?

### DISCUSSION THREAD

**Discussion by mariano-daniel** [Homebrew/discussions#3909] [Getting started]
Hello everyone! I'm trying to find packages by description, but I'm unable to pull it right:

`brew -S --desc --eval-all /Sofpthone/`

I tried with `brew search --desc --eval-all Sofpthone *`but still nothing.

How do I do? 

ALSO; how do I make the search case insensitive? `man brew`does not explain this.

Thanks in advance!

**Comment by cho-m** [comment]
I think there is a bug/limitation in search where the `--eval-all` doesn't update search cache and you currently need to use environment variable `HOMEBREW_EVAL_ALL` instead (at least for initial cache update).

Also, there is a typo in your search "Sofpthone". I don't think fuzzy search is supported.

e.g.
```console
❯ brew -S --desc --eval-all /Softphone/
==> Formulae

==> Casks

❯ HOMEBREW_EVAL_ALL=1 brew -S --desc /Softphone/
==> Formulae

==> Casks
bria: (Bria) Softphone application
sipgate-softphone: (Sipgate Softphone) Make telephone calls on the computer

❯ brew -S --desc --eval-all /Softphone/
==> Formulae

==> Casks
bria: (Bria) Softphone application
sipgate-softphone: (Sipgate Softphone) Make telephone calls on the computer
```

---

Homebrew uses Ruby so the regex input should match behavior of Ruby's Regexp.

For case-insensitivity, since the command needs to end in a slash, you can't use the syntax: `/.../i`. Instead, you might need to use subexpression construct `/(?i:...)/`
```console
❯ HOMEBREW_EVAL_ALL=1 brew -S --desc /Softphone/i
==> Formulae

==> Casks

❯ HOMEBREW_EVAL_ALL=1 brew -S --desc '/(?i:Softphone)/'
==> Formulae

==> Casks
bria: (Bria) Softphone application
jami: (Jami, Savoir-faire Linux Ring) Decentralised instant messenger and softphone
sipgate-softphone: (Sipgate Softphone) Make telephone calls on the computer
```

Ref: https://ruby-doc.org/core-2.6.10/Regexp.html#class-Regexp-label-Options

**Reply by cho-m** [reply]
Related issue: https://github.com/Homebrew/brew/issues/14010
